File: python/typical90_bw.py
n=int(input())
l=[]
from collections import defaultdict
d=defaultdict(int)
for i in range(2,n+1):
	if i*i>n:break
	while n%i==0:
		n//=i
		d[i]+=1
if n!=1:d[n]+=1
t=sum(d.values())
x=1
ans=0
while x<t:
	x*=2
	ans+=1
print(ans)

File: rust/typical90/src/bin/typical90_bw.rs
use proconio::input;
fn main() {
	input!{n:usize}
	let mut x = n;
	let mut cnt = 0;
	for i in 2..n {
		if i*i>n {break}
		while x%i==0 {
			x /= i;
			cnt += 1;
		}
	}
	if x!=1 {cnt+=1}
	let mut ans = 0;
	let mut i = 1;
	while i<cnt {
		ans += 1;
		i *= 2;
	}
	println!("{}",ans);
}
